### DISCUSSION TITLE
Do something when user clicked the 'Reset Filters' button of a Table

### DISCUSSION THREAD

**Discussion by kuchienkz** [filamentphp/filament#10080] [Help] [answered]
I'm using `updated` Livewire hook to update MyWidget view when user made changes to a Table:

```
    // Somewhere in ListData.php (of DataResource)
    public function updated($name)
    {
        if (Str::of($name)->contains(['mountedTableAction', 'mountedTableBulkAction', 'tableFilters'], true)) {
            // Update MyWidget
        }
    }
```
MyWidget is properly updated when user did CRUD operations and updated the filter form, but MyWidget is not updating when user clicked the "Reset Filters" button (below filter form).

How can I update MyWidget when user clicked the "Reset Filters" button?

**Comment by vitalijalbu** [comment]
will be great !!!

**Comment by leandrocfe** [comment, marked as answer]
I think you can add this in the ListPage

```php
public function resetTableFiltersForm(): void
{
    parent::resetTableFiltersForm();
    $this->dispatch('your-custom-event');
}
```

**Reply by kuchienkz** [reply]
Work as expected. Thanks!

**Comment by bikash-s-bhandari** [comment]
@leandrocfe  how to change table filter reset button from link() to button(), and how to keep reset and apply button together?

**Reply by leandrocfe** [reply]
This is here: https://github.com/filamentphp/filament/blob/3.x/packages/tables/resources/views/components/filters/index.blade.php

You could publish this view (this isn't recommended)
